libs/auth/src/dipped_smart_card_auth.ts

```diff
@@ -127,22 +129,27 @@ async function logAuthEventIfNecessary(
           newAuthStatus.wrongPinEnteredAt !==
             previousAuthStatus.wrongPinEnteredAt
         ) {
-          await logger.log(
-            LogEventId.AuthPinEntry,
-            previousAuthStatus.user.role,
-            {
-              disposition: LogDispositionStandardTypes.Failure,
-              message: 'User entered incorrect PIN.',
-            }
-          );
+          await logger.log(LogEventId.AuthPinEntry, newAuthStatus.user.role, {
+            disposition: LogDispositionStandardTypes.Failure,
+            message: 'User entered incorrect PIN.',
+          });
+        } else if (
+          newAuthStatus.error &&
+          newAuthStatus.error.erroredAt !== previousAuthStatus.error?.erroredAt
+        ) {
+          await logger.log(LogEventId.AuthPinEntry, newAuthStatus.user.role, {
+            disposition: LogDispositionStandardTypes.Failure,
+            message: `Error checking PIN: ${extractErrorMessage(
+              newAuthStatus.error.error
+            )}`,
+          });
         }
       } else if (newAuthStatus.status === 'remove_card') {
         await logger.log(LogEventId.AuthPinEntry, newAuthStatus.user.role, {
           disposition: LogDispositionStandardTypes.Success,
           message: 'User entered correct PIN.',
         });
       }
-      // PIN check errors are logged in checkPin, where we have access to the full error message
```

Comment:
The main goal of this commit was to be able to remove this comment / exception-to-the-rule log